hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go

```diff
@@ -90,7 +90,7 @@ func (a Azure) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _ *hy
 					Args: []string{
 						"--namespace=$(MY_NAMESPACE)",
 						"--leader-elect=true",
-						"--feature-gates=MachinePool=false",
+						"--feature-gates=MachinePool=false,ASOAPI=false",
```

Comment:
Per a discussion on the Kubernetes Slack, turning this off doesn't disable ASO; instead, it uses a different ASO API.

https://kubernetes.slack.com/archives/CEX9HENG7/p1731607828878759?thread_ts=1731602144.648199&cid=CEX9HENG7

Reply:
Apparently that message was incorrect. Going to put this into our CAPZ deployment, too.